File: DemoMethods/Basic/Query1.cs
using System;
using System.Linq;
using System.Web.Http;
using DemoMethods.Entities;
using DemoMethods.Helpers;
using Raven.Client;
using Raven.Client.Linq;

namespace DemoMethods.Basic
{
    public partial class BasicController : DemoApiController
    {

        [HttpGet]
        [Demo("Query I", DemoOutputType.Flatten)]
        public object Query1(string country = "UK")
        {
            using (var session = DocumentStoreHolder.Store.OpenSession())
            {
                RavenQueryStatistics stats;
                var q = 
                    from company in session.Query<Company>().Statistics(out stats)
                    where company.Address.Country == country
                    select company;

                var result = q.FirstOrDefault();
                return new
                {
                    stats.IndexName,
                    Result = result
                };
            }
        }
    }
}

File: DemoServer/PythonDemo/demo_methods/Basic/Query1.py
from demo.holder import DocumentStoreHolder
from demo.entities import *


class Query1(object):
    @staticmethod
    def query1(country="UK"):
        with DocumentStoreHolder.get_store().open_session() as session:
            result, stats = list(session.query(object_type=Company,
                                               with_statistics=True).
                                 where(**{"Address.Country": country}))

            return result[0] if result else None
